src/main/webapp/app/shared/util/download.util.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpResponse } from '@angular/common/http';
+
+export function downloadZipFileFromResponse(response: HttpResponse<Blob>): void {
+    if (response.body) {
+        const file = new Blob([response.body], { type: 'application/zip' });
+        downloadFile(file, response.headers.get('filename')!);
+    }
+}
+
+/**
+ * Make a file of given blob and allows user to download it from the browser.
+ * @param blob data to be written in file.
+ * @param filename suggested to the browser.
+ */
+export function downloadFile(blob: Blob, filename: string) {
```

Comment:
@krusche I refactored your new method in #1995 and what was used in `quiz-exercise.service.ts` here as this is duplicate. This here drops the `$window` argument though.